File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    bool solve(int i, int j, string &s) {             // recursion palindrome check
        if (i >= j) {
            return true;
        }
        if (s[i] == s[j]) {
            return solve(i+1, j-1, s);  
        } 
        else {
            return false;
        }
    }

    string longestPalindrome(string s) {
        int Maxlen = 0;  
        int sp = 0; // starting ppont


        for (int i = 0; i < s.length(); i++) {

            for (int j = i; j < s.length(); j++) {                  // sare substring dekh lenge

                if (solve(i, j, s) == true) {
                    
                    if (j - i + 1 > Maxlen) {
                        Maxlen = j - i + 1;         // maleng update
                        sp = i;                     // starting point ka ttrack
                    }
                }
            }
        }
        return s.substr(sp, Maxlen);  // starting points leke end tak substring return
    }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public boolean isPalindrome(int i, int j, String s) {
        while(i<j){
            if(s.charAt(i)!= s.charAt(j)){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }

    public String longestPalindrome(String s) {
        
        int maxLen = 0;
        int start = 0;

        for (int i = 0; i < s.length(); i++) {
            for (int j = i; j < s.length(); j++) {

                if (isPalindrome(i, j, s)) {
                    if (j - i + 1 > maxLen) {
                        maxLen = j - i + 1;
                        start = i;
                    }
                }
            }
        }

        return s.substring(start,  start+maxLen);
    }
}
